src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -40149,6 +40151,416 @@ public void CopyTupleUnconstrainedElementNullability_02()
+        x.Value.ToString();
+        y.Value.ToString(); // 1
+        z.Value.ToString();
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (9,9): warning CS8629: Nullable value type may be null.
+                //         y.Value.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "y.Value").WithLocation(9, 9));
+        }
+
+        [Fact]
+        [WorkItem(32531, "https://github.com/dotnet/roslyn/issues/32531")]
+        public void Conversions_ExplicitNullable_05()
+        {
+            var source =
+@"class Program
+{
+    static void F1(int? x1)
+    {
+        int y1 = (int)x1; // 1
+    }
+    static void F2(int? x2)
+    {
+        if (x2 == null) return;
+        int y2 = (int)x2;
+    }
+    static void F3(int? x3)
+    {
+        long y3 = (long)x3; // 2
+    }
+    static void F4(int? x4)
+    {
+        if (x4 == null) return;
+        long y4 = (long)x4;
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (5,18): warning CS8629: Nullable value type may be null.
+                //         int y1 = (int)x1; // 1
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "(int)x1").WithLocation(5, 18),
+                // (14,19): warning CS8629: Nullable value type may be null.
+                //         long y3 = (long)x3; // 2
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "(long)x3").WithLocation(14, 19));
+        }
+
+        [Fact]
+        [WorkItem(32531, "https://github.com/dotnet/roslyn/issues/32531")]
+        public void Tuple_Conversions_ImplicitNullable()
+        {
+            var source =
+@"struct S
+{
+    internal object? F;
+}
+class Program
+{
+    static void F()
+    {
+        S x = new S();
+        S y = new S() { F = 1 };
+        (S?, S) t = (x, y);
+        (S? a, S? b) u = t;
+        S t1 = t.Item1.Value;
+        S t2 = t.Item2;
+        t1.F.ToString(); // 1
+        t2.F.ToString();
+        S u1 = u.a.Value;
+        S u2 = u.b.Value;
+        u1.F.ToString(); // 2
+        u2.F.ToString();
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            // https://github.com/dotnet/roslyn/issues/32599: Support implicit tuple conversions. See // 1, 2.
+            comp.VerifyDiagnostics(
+                // (18,16): warning CS8629: Nullable value type may be null.
+                //         S u2 = u.b.Value;
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "u.b.Value").WithLocation(18, 16));
+        }
+
+        [Fact]
+        [WorkItem(32599, "https://github.com/dotnet/roslyn/issues/32599")]
+        public void Tuple_Conversions_ImplicitReference()
+        {
+            var source =
+@"class Program
+{
+    static void F(string x, string? y)
+    {
```

Comment:
I'm unsure about this. 
In the view that `(x, y)`  means `new ValueTuple<string, string?>(x, y)`, I would expect a warning on the conversion (like `Pair<string, string?>` to `Pair<object?, string?>`).

Reply:
Yes, I understand, but that means we're not analyzing tuples like we are analyzing `Pair<T, T>`. 
I'm trying to clarify my mental model for analyzing tuples, but it seems we're taking different approaches in different contexts.